handler/src/main/java/io/netty/handler/ssl/JdkAlpnSslEngine.java

```diff
@@ -62,12 +60,15 @@ private static void updateAvailability() {
                     "protocolSelector");
             ALPN.put(engine, new ServerProvider() {
                 @Override
-                public String select(List<String> protocols) {
+                public String select(List<String> protocols) throws SSLException {
                     try {
                         return protocolSelector.select(protocols);
+                    } catch(SSLException e) {
+                        throw e;
                     } catch (Throwable t) {
-                        PlatformDependent.throwException(t);
-                        return null;
+                        // Ensure that all exceptions are propagated as SSLExceptions
+                        // so that the SslHandler properly fails the handshake.
+                        throw new SSLException(t);
```

Comment:
It's an interesting question. I'm thinking that it wouldn't actually matter too much, though. The catching code generally only cares about `SSLException` in general.

The argument for using `SSLHandshakeException`, in my mind, is that a failure occurred while handshaking. The argument for using `SSLException` is that handshake's documentation says it "Indicates that the client and server could not negotiate the desired level of security." Using `SSLHandshakeException` in `JdkBaseApplicationProtocolNegotiator` would then be a stretch, and wrapping an arbitrary exception here seems beyond tenuous.

I'm on the fence, but am leaning more toward SSLException.